src/views/Leads/GridItem/index.js

```diff
@@ -0,0 +1,298 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+import { Link } from 'react-router-dom';
+import classNames from 'classnames';
+import Cloak from '#components/Cloak';
+import DetectOutsideClick from '#components/DetectOutsideClick';
+import Button from '#rsca/Button';
+import Modal from '#rscv/Modal';
+import LeadPreview from '#components/LeadPreview';
+import DangerConfirmButton from '#rsca/ConfirmButton/DangerConfirmButton';
+import {
+    iconNames,
+    pathNames,
+} from '#constants';
+import { reverseRoute } from '#rsu/common';
+import {
+    leadTypeIconMap,
+    leadPaneTypeMap,
+    LEAD_PANE_TYPE,
+} from '#entities/lead';
+import _ts from '#ts';
+import { timeFrom } from '#utils/common';
+
+import leadThumbnail from '#resources/img/lead-thumbnail.png';
+import styles from './styles.scss';
+
+const propTypes = {
+    className: PropTypes.string,
+    lead: PropTypes.object.isRequired, // eslint-disable-line react/forbid-prop-types
+    style: PropTypes.object, // eslint-disable-line react/forbid-prop-types
+    activeProject: PropTypes.number.isRequired,
+    onSearchSimilarLead: PropTypes.func.isRequired,
+    onRemoveLead: PropTypes.func.isRequired,
+    onMarkProcessed: PropTypes.func.isRequired,
+    onMarkPending: PropTypes.func.isRequired,
+    minHeight: PropTypes.number, // eslint-disable-line react/no-unused-prop-types
+};
+
+const defaultProps = {
+    className: '',
+    style: {},
+    minHeight: 295,
+};
+
+export default class GridItem extends React.PureComponent {
+    static propTypes = propTypes;
+    static defaultProps = defaultProps;
+
+    // for masonry
+    static getColumnSpanFromProps = () => 1
+
+    static getHeightFromProps = (getState, props) => {
+        const {
+            lead: {
+                thumbnailHeight = 400, thumbnailWidth = 250,
+            }, width = 250, minHeight = 295,
+        } = props;
+        // item is wrapped by masonry
+        const imgHeight = Math.max((width * thumbnailHeight) / thumbnailWidth, minHeight);
+        return imgHeight;
+    }
+
+    state = {
+        showPreview: false,
+    }
+
+    get links() {
+        const { activeProject, lead } = this.props;
+        return {
+            editLead: {
+                pathname: reverseRoute(
+                    pathNames.addLeads,
+                    {
+                        projectId: activeProject,
+                    },
+                ),
+                state: {
```

Comment:
the state attribute from links seems to be unused.
IMO these computations could be inlined